zeus/device/gpu/amd.py

```diff
@@ -241,29 +251,49 @@ def getAverageMemoryPowerUsage(self) -> int:
         )
 
     @_handle_amdsmi_errors
-    def supportsGetTotalEnergyConsumption(self) -> bool:
-        """Check if the GPU supports retrieving total energy consumption."""
-        if self._supportsGetTotalEnergyConsumption is None:
-            try:
-                _ = amdsmi.amdsmi_get_energy_count(self.handle)
-                self._supportsGetTotalEnergyConsumption = True
-            except amdsmi.AmdSmiLibraryException as e:
-                if (
-                    e.get_error_code() == 2
-                ):  # amdsmi.amdsmi_wrapper.AMDSMI_STATUS_NOT_SUPPORTED
-                    self._supportsGetTotalEnergyConsumption = False
-                else:
-                    raise e
-
+    def supportsGetTotalEnergyConsumption(
+        self,
+    ) -> bool:
+        """Check if the GPU supports retrieving total energy consumption. Returns a future object of the result."""
+        wait_time = 0.5  # seconds
+        threshold = 0.8  # 80% threshold
+
+        power = self.getInstantPowerUsage()
+        initial_energy = self.getTotalEnergyConsumption()
+        time.sleep(wait_time)
+        final_energy = self.getTotalEnergyConsumption()
+
+        measured_energy = final_energy - initial_energy
+        expected_energy = power * wait_time  # power is in mW, wait_time is in seconds
+
+        # if the difference between measured and expected energy is less than 1% of the expected energy, then the API is supported
+        if abs(measured_energy - expected_energy) < threshold * expected_energy:
+            self._supportsGetTotalEnergyConsumption = True
+        else:
+            self._supportsGetTotalEnergyConsumption = False
+            logger.info(
+                "Disabling `getTotalEnergyConsumption` for device %d. The result of `amdsmi.amdsmi_get_energy_count` is not accurate. Expected energy: %d mJ, Measured energy: %d mJ"
+                "This is a known issue with some AMD GPUs, please see https://github.com/ROCm/amdsmi/issues/38 for more information."
+                "Energy metrics will still be available and measured through polling of `getInstantPowerUsage` method.",
+                self.gpu_index,
+                expected_energy,
+                measured_energy,
+            )
         return self._supportsGetTotalEnergyConsumption
 
     @_handle_amdsmi_errors
     def getTotalEnergyConsumption(self) -> int:
         """Return the total energy consumption of the GPU since driver load. Units: mJ."""
-        info = amdsmi.amdsmi_get_energy_count(self.handle)
-        return int(
-            info["power"] / 1e3
-        )  # returns in micro Joules, convert to mili Joules
+        energy_dict = amdsmi.amdsmi_get_energy_count(self.handle)
+        if "energy_accumulator" in energy_dict:  # New API
```

Comment:
```suggestion
        if "energy_accumulator" in energy_dict:  # Changed since amdsmi 6.2.1
```

Reply:
Might exceed line limit, please adapt as appropriate. Just wanted to document when this API change was made.